src/lib/sessions.ts

```diff
@@ -44,16 +45,22 @@ export const saveSession = (session: Session): void => {
 	const currentSessions = get(sessionsStore) || [];
 
 	// Find the index of the session with the same id, if it exists
-	const existingSessionIndex = currentSessions.findIndex(s => s.id === session.id);
+	const existingIndex = currentSessions.findIndex((k) => k.id === session.id);
 
-	if (existingSessionIndex !== -1) {
+	if (existingIndex !== -1) {
 		// Update the existing session
-		currentSessions[existingSessionIndex] = session;
+		currentSessions[existingIndex] = session;
 	} else {
 		// Add the new session if it doesn't exist
 		currentSessions.push(session);
 	}
 
-	// Update the store, which will trigger the localStorage update
-	sessionsStore.set(currentSessions);
+	// Sort the sessions by updatedAt in descending order (most recent first)
+	const sortedSessions = currentSessions.sort((a, b) => {
+		if (!a.updatedAt || !b.updatedAt) return 0;
+		return new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime();
+	});
```

Comment:
This _mostly works_ but there's an edge-case bug I found where if `a.updatedAt` exists and `b.updatedAt` is `undefined` then the guard clause on line 60 returns `0`, but that's incorrect because that condition should be sorted.

Also, the guard clause is missing in `saveKnowledge()`, but because those sorting functions will be identical I think it might be time to DRY these save functions altogether.

Can you try adding a new function on `src/lib/store.ts` called something like `function saveAndSortStore()` that abstracts the logic in `saveKnowledge()` and `saveSession()` which is entirely identical at this point.

This will be a good blueprint to eventually DRY'ing `loadKnowledge()` and `loadSession()` but we'll punt that for later.

Reply:
About this, I've been thinking: currently, the saveSession and saveKnowledge functions receive one param, the updated item.

An idea was to implement the function with two params, the store to update and the updated item (of generic type T).

The other idea was to keep the only one param (the item) and decide which store to update based in some attribute of the item, but I'm not so sure that this is a good idea.

What do you think?

Also, maybe we could extract this DRY to a separated issue and close this one for now.

Reply:
Yeah, good point.

Let's just DRY the sorting function and keep everything else the same and can optimize the rest later.